Do-It/Controller/CreateDoItTableViewController.swift

```diff
@@ -95,9 +115,16 @@ class CreateDoItTableViewController: UITableViewController {
     }
 
     @IBAction func save(_ sender: Any) {
-        let doIt = DoIt(course: course, dueDate: date, description: desc, name: name,
+        var doIt = DoIt(course: course, dueDate: date, description: desc, name: name,
                         priority: priority, kind: .homework)
-        delegate?.createDoItViewController(self, didSaveDoIt: doIt)
+
+        if let doItToEdit = doItToEdit {
```

Comment:
Then, this `if let` will become a `switch` statement where you can get the editing do-it via pattern matching.